### DISCUSSION TITLE
Rule as array or string

### DISCUSSION THREAD

**Discussion by meteoguzhan** [adonisjs/core#4010] [Help]
Hello everyone,
How can I add rule as array or string in validation operation?
My current codes:
```
export const CreationParams = schema.create({
  value: schema.string() or schema.array(),
})
```

**Comment by tomgobich** [comment]
Unfortunately, that's not currently possible with the Validator, but it is something that's been acknowledged as missing.

In your case, however, you do have a couple of options. (I haven't tested either of the below, so they may not be exact, but should be close to working examples)

1. You could create two validators, one for string and another for array
```ts
const stringSchema = schema.create({ value: schema.string.optional() }) // <- optional to bypass error if not provided
let data = await request.validate({ schema: stringSchema })

if (!data.value) {
  const arraySchema = schema.create({ value: schema.array() })
  data = await request.validate({ schema: arraySchema })
}

const value = data.value
```

2. You could check for a string value prior to validation and mutate it to an array, then just validate for an array
```ts
// I'd put this portion in a service method 
const body = request.body()

if (typeof body.value === 'string') {
  body.value = [body.value] // convert to array
}

request.updateBody(body)
// end

// then you should be able to just validate for an array
const arraySchema = schema.create({ value: schema.array() })
const data = await request.validate({ schema: arraySchema })
```

3. Depending on how your HTML is, this may be something relatively easily fixed on the front end by converting the field to an array type by suffixing the name with `[]`. For example, `taxonomies[]`, which will send the form value as an array.
